app/models/spree/calculator/avalara_transaction.rb

```diff
@@ -64,7 +64,7 @@ def get_avalara_response(order)
 
     def long_cache_key(order)
       key = order.avatax_cache_key
-      key << order.tax_address.try(:cache_key).to_s
+      key << order.tax_address.try(:avatax_cache_key).to_s
```

Comment:
:+1: it's important we keep this as `try` not `try!`, etc. If the order doesn't have a `ship_address` set, `order.tax_address` will fall back to `order.store.default_cart_tax_location` which doesn't have a `[avatax_]cache_key` method.

Reply:
Gotcha! You mentioned that earlier so I didn't change it!